### PR TITLE
Docs for some nodes in visual shader(part 3)

### PR DESCRIPTION
**Pull request by Chaosus**
Docs for:

- VisualShaderNodeFloatConstant
- VisualShaderNodeFloatFunc
- VisualShaderNodeFloatOp
- VisualShaderNodeIf
- VisualShaderNodeInput
- VisualShaderNodeIntConstant
- VisualShaderNodeIntFunc
- VisualShaderNodeIntOp

### CODE REVIEW

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 1]
```diff
@@ -1,81 +1,116 @@
...
 	<methods>
 	</methods>
 	<members>
 		<member name="function" type="int" setter="set_function" getter="get_function" enum="VisualShaderNodeFloatFunc.Function" default="13">
+			A function to be applied to the scalar. See [enum Function] for options.
 		</member>
 	</members>
 	<constants>
 		<constant name="FUNC_SIN" value="0" enum="Function">
+			Returns the sine of the parameter. Translates to [code]sin(x)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: in _the_ Godot Shader Language

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 2]
```diff
@@ -1,81 +1,116 @@
...
 		</constant>
 		<constant name="FUNC_COS" value="1" enum="Function">
+			Returns the cosine of the parameter. Translates to [code]cos(x)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: in the Godot Shader Language

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 3]
```diff
@@ -1,81 +1,116 @@
...
 		</constant>
 		<constant name="FUNC_TAN" value="2" enum="Function">
+			Returns the tangent of the parameter. Translates to [code]tan(x)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: in _the_ Godot Shader Language

Same for below mentions of Godot Shader Language

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 4]
```diff
@@ -1,81 +1,116 @@
...
+			Returns the inverse of the square root of the parameter. Translates to [code]inversesqrt(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_LOG2" value="26" enum="Function">
+			Returns the base 2 logarithm of the parameter. Translates to [code]log2(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_RADIANS" value="27" enum="Function">
+			Convert a quantity in degrees to radians. Translates to [code]radians(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_RECIPROCAL" value="28" enum="Function">
+			Finds reciprocal value of dividing 1 on [code]x[/code] by [code]1 / x[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: Finds reciprocal value of dividing 1 _by_ [code]x[/code] (i.e. [code]1 / x[/code]).

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 5]
```diff
@@ -1,81 +1,116 @@
...
 		</constant>
 		<constant name="FUNC_ROUNDEVEN" value="29" enum="Function">
+			Finds the nearest even integer to the parameter. Translates to [code]roundEven(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_TRUNC" value="30" enum="Function">
+			Returns a value equal to the nearest integer to [code]x[/code] whose absolute value is not larger than the absolute value of [code]x[/code]. Translates to [code]trunc(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_ONEMINUS" value="31" enum="Function">
+			Subtracts scalar from 1 by [code]1 - x[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: Subtracts scalar [code]x[/code] from 1 (i.e. [code]1 - x[/code]).

File: doc/classes/VisualShaderNodeFloatOp.xml [hunk 6]
```diff
@@ -1,37 +1,50 @@
...
 	<methods>
 	</methods>
 	<members>
 		<member name="operator" type="int" setter="set_operator" getter="get_operator" enum="VisualShaderNodeFloatOp.Operator" default="0">
+			An operator to be applied to the inputs. See [enum Operator] for options.
 		</member>
 	</members>
 	<constants>
 		<constant name="OP_ADD" value="0" enum="Operator">
+			Sums two numbers by [code]a + b[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatOp.xml]: Sums two numbers using [code]a + b[/code].

File: doc/classes/VisualShaderNodeFloatOp.xml [hunk 7]
```diff
@@ -1,37 +1,50 @@
...
 		</constant>
 		<constant name="OP_SUB" value="1" enum="Operator">
+			Subtracts two numbers by [code]a - b[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatOp.xml]: Subtracts two numbers using [code]a - b[/code].

File: doc/classes/VisualShaderNodeFloatOp.xml [hunk 8]
```diff
@@ -1,37 +1,50 @@
...
 		</constant>
 		<constant name="OP_MUL" value="2" enum="Operator">
+			Multiplies two numbers by [code]a * b[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatOp.xml]: Multiplies two numbers using [code]a * b[/code].

File: doc/classes/VisualShaderNodeFloatOp.xml [hunk 9]
```diff
@@ -1,37 +1,50 @@
...
 		</constant>
 		<constant name="OP_DIV" value="3" enum="Operator">
+			Divides two numbers by [code]a / b[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatOp.xml]: Divides two numbers using [code]a / b[/code].

File: doc/classes/VisualShaderNodeIntOp.xml [hunk 10]
```diff
@@ -1,31 +1,41 @@
...
 	<methods>
 	</methods>
 	<members>
 		<member name="operator" type="int" setter="set_operator" getter="get_operator" enum="VisualShaderNodeIntOp.Operator" default="0">
+			An operator to be applied to the inputs. See [enum Operator] for options.
 		</member>
 	</members>
 	<constants>
 		<constant name="OP_ADD" value="0" enum="Operator">
+			Sums two numbers by [code]a + b[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeIntOp.xml]: See above regarding in *floatOp*

**Review by clayjohn — CHANGES_REQUESTED** [review]
Looks great!

All references to "Godot Shader Language" should be to "_the_ Godot Shader Language"

Also note changes to "... by [code]...[/code] formula."

The suggested changes should be applied to all similar instances.

**Comment by Chaosus** [comment]
@clayjohn Done.. check again pls

File: doc/classes/VisualShaderNodeIntFunc.xml [hunk 11]
```diff
@@ -1,25 +1,32 @@
...
+			Returns the absolute value of the parameter. Translates to [code]abs(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_CLAMP" value="1" enum="Function">
+			Constrains a parameter between [code]min[/code] and [code]max[/code]. Translates to [code]clamp(x, min, max)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_NEGATE" value="2" enum="Function">
+			Negates the [code]x[/code] by [code]-(x)[/code] formula.
 		</constant>
 		<constant name="FUNC_SIGN" value="3" enum="Function">
+			Extracts the sign of the parameter. Translates to [code]sign(x)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeIntFunc.xml]: Missed one

File: doc/classes/VisualShaderNodeInput.xml [hunk 12]
```diff
@@ -11,16 +12,19 @@
 			<return type="String">
 			</return>
 			<description>
+				Returns a translated name of the current constant in Godot Shader Language eg. [code]"ALBEDO"[/code] if the [member input_name] equal to [code]"albedo"[/code].
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeInput.xml]: _the_ Godot

File: doc/classes/VisualShaderNodeIntFunc.xml [hunk 13]
```diff
@@ -1,25 +1,32 @@
...
 		<member name="function" type="int" setter="set_function" getter="get_function" enum="VisualShaderNodeIntFunc.Function" default="0">
+			A function to be applied to the scalar. See [enum Function] for options.
 		</member>
 	</members>
 	<constants>
 		<constant name="FUNC_ABS" value="0" enum="Function">
+			Returns the absolute value of the parameter. Translates to [code]abs(x)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_CLAMP" value="1" enum="Function">
+			Constrains a parameter between [code]min[/code] and [code]max[/code]. Translates to [code]clamp(x, min, max)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeIntFunc.xml]: Missed one

File: doc/classes/VisualShaderNodeIntFunc.xml [hunk 14]
```diff
@@ -1,25 +1,32 @@
...
 	<methods>
 	</methods>
 	<members>
 		<member name="function" type="int" setter="set_function" getter="get_function" enum="VisualShaderNodeIntFunc.Function" default="0">
+			A function to be applied to the scalar. See [enum Function] for options.
 		</member>
 	</members>
 	<constants>
 		<constant name="FUNC_ABS" value="0" enum="Function">
+			Returns the absolute value of the parameter. Translates to [code]abs(x)[/code] in Godot Shader Language.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeIntFunc.xml]: Missed one

File: doc/classes/VisualShaderNodeIntFunc.xml [hunk 15]
```diff
@@ -1,25 +1,32 @@
...
 		</constant>
 		<constant name="FUNC_CLAMP" value="1" enum="Function">
+			Constrains a parameter between [code]min[/code] and [code]max[/code]. Translates to [code]clamp(x, min, max)[/code] in Godot Shader Language.
 		</constant>
 		<constant name="FUNC_NEGATE" value="2" enum="Function">
+			Negates the [code]x[/code] by [code]-(x)[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeIntFunc.xml]: Negates the [code]x[/code] using [code]-(x)[/code].

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 16]
```diff
@@ -1,81 +1,116 @@
...
+			Finds the nearest integer that is greater than or equal to the parameter. Translates to [code]ceil(x)[/code] in the Godot Shader Language.
 		</constant>
 		<constant name="FUNC_FRAC" value="17" enum="Function">
+			Computes the fractional part of the argument. Translates to [code]fract(x)[/code] in the Godot Shader Language.
 		</constant>
 		<constant name="FUNC_SATURATE" value="18" enum="Function">
+			Clamps the value between [code]0.0[/code] and [code]1.0[/code] by [code]min(max(x, 0.0), 1.0)[/code] formula.
 		</constant>
 		<constant name="FUNC_NEGATE" value="19" enum="Function">
+			Negates the [code]x[/code] by [code]-(x)[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: Negates the [code]x[/code] using [code]-(x)[/code].

File: doc/classes/VisualShaderNodeFloatFunc.xml [hunk 17]
```diff
@@ -1,81 +1,116 @@
...
+			Finds the nearest integer to the parameter. Translates to [code]round(x)[/code] in the Godot Shader Language.
 		</constant>
 		<constant name="FUNC_CEIL" value="16" enum="Function">
+			Finds the nearest integer that is greater than or equal to the parameter. Translates to [code]ceil(x)[/code] in the Godot Shader Language.
 		</constant>
 		<constant name="FUNC_FRAC" value="17" enum="Function">
+			Computes the fractional part of the argument. Translates to [code]fract(x)[/code] in the Godot Shader Language.
 		</constant>
 		<constant name="FUNC_SATURATE" value="18" enum="Function">
+			Clamps the value between [code]0.0[/code] and [code]1.0[/code] by [code]min(max(x, 0.0), 1.0)[/code] formula.
```
Comment (clayjohn) [inline doc/classes/VisualShaderNodeFloatFunc.xml]: Clamps the value between [code]0.0[/code] and [code]1.0[/code] using [code]min(max(x, 0.0), 1.0)[/code].

**Review by clayjohn — CHANGES_REQUESTED** [review]
Just a few more of the same we missed. :)

**Comment by Chaosus** [comment]
Done.

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**